File: StayBookingFrontend/lib/model/update_room_request_dto.dart
import 'package:stay_booking_frontend/model/room_upload_file.dart';

class UpdateRoomRequestDto {
  UpdateRoomRequestDto({
    required this.hotelId,
    required this.roomType,
    required this.description,
    required this.price,
    required this.available,
    required this.updatedBy,
    required this.photoFiles,
    required this.replacePhotos,
  });

  final int hotelId;
  final String roomType;
  final String description;
  final double price;
  final bool available;
  final String updatedBy;
  final List<RoomUploadFile> photoFiles;
  final bool replacePhotos;
}

File: StayBooking/StayBooking/src/main/java/com/dto/UpdateRoomRequestDto.java
package com.dto;

import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import java.math.BigDecimal;
import java.util.List;

@Data
public class UpdateRoomRequestDto {
    private Integer hotelId;
    private String hotelName;
    private String roomType;
    private String description;
    private BigDecimal price;
    private Boolean available;
    private Boolean replacePhotos;
    private String updatedby;
    private List<MultipartFile> photos;
}
